### ISSUE TITLE
License Contamination

### ISSUE THREAD

**Issue by y-takshima** [fzaninotto/Faker#881]
Based on the LICENSE and the CONTRIBUTING.md, the software seems to be under the MIT license. But when I search "license" string within this repository, there are some files described as a creative commons.

https://github.com/fzaninotto/Faker/search?utf8=%E2%9C%93&q=CREATIVE+COMMONS

Should I consider all the repository files under the MIT license or not?

**Comment by Newman101** [comment]
Looks like some of the texts are being taken from Ivan Franko's poem "Zakhar Berkut". Ivan Franko died 1916 so as such, at least that text should be under public domain. If for some reason it's not under public domain, then the text can be replaced with a text which is.

**Comment by fzaninotto** [comment]
No, some portions are Creative Commons, although they all concern public domain novels. Do you have any advice on how to present that?

**Comment by y-takshima** [comment]
I believe that it is better not to use CC BY-SA content inside OSS. It may not be re- distributed as the MIT license since ShareAlike terms affects.

> ShareAlike
 — If you remix, transform, or build upon the material, you must distribute your contributions under the same license as the original.

**Comment by okj579** [comment]
It is possible to distribute software under a mixed license. However, it would probably be easier to make sure everything is either public domain or MIT.

The Khazakh text is a Wikipedia article and truly CC. It should be probably be replaced anyway.

The Russian and Ukranian texts are in the public domain. They were probably copied from Wikipedia/Wikisource, but that doesn't matter if they don't have any effective copyright themselves. The copyright notices can be removed, as they are incorrect.

The Farsi text is fairly recent and is not public domain (till at least 2019 when it passes into the public domain in Iran and some other countries). As far as I can tell, it has not been released under CC or any other permissive license, so its current use on Wikisource and here wouldn't be legal.

**Comment by fzaninotto** [comment]
Fine by me. Could you open a pull request with these changes ?

**Comment by Newman101** [comment]
I can make the PR. Before that, I need to google some public domain resources for Kazakh and Farsi texts. Any ideas where to start?

**Comment by Newman101** [comment]
Also, are there any length requirements?

**Comment by okj579** [comment]
I started looking and found a book by Abay Qunanbayuli (Kazakh) on
Wikisource. So far, I've been unable to find modern Persian prose old
enough to be in the public domain.

**Comment by fzaninotto** [comment]
If you can't find any, then simply remove the one published in Faker.

**Comment by Newman101** [comment]
After sorting ```kk_KZ``` locale out, I'll do some investigation for Farsi and if found a reasonable PD prose, I'll use that. If not, then just deleting the whole thing.

**Comment by Newman101** [comment]
Last thing to do: I guess the copyright notices from Ukrainian and Russian texts can be removed?

**Comment by fu7mu4** [comment]
Text.php of it_IT is released under CC BY-NC-SA, too.
https://github.com/fzaninotto/Faker/search?utf8=%E2%9C%93&q=Attribution-NonCommercial-ShareAlike&type=

**Comment by fzaninotto** [comment]
Pinocchio is public domain anyway. I think the copyright notice can be removed from this one.
